src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -142,29 +142,46 @@ internal double ConvertMoerToGramsPerKilowattHour(double value)
         return value * LBS_TO_GRAMS_CONVERSION_FACTOR / MWH_TO_KWH_CONVERSION_FACTOR;
     }
 
-    private IEnumerable<EmissionsData> ConvertToEmissionsData(IEnumerable<GridEmissionDataPoint> data)
+    private IEnumerable<EmissionsData> ConvertToEmissionsData(IEnumerable<GridEmissionDataPoint> gridEmissionDataPoints)
     {
+        var defaultDuration = GetDurationFromGridEmissionDataPointsOrDefault(gridEmissionDataPoints, TimeSpan.Zero);
+        
         // Linq statement to convert WattTime forecast data into EmissionsData for the CarbonAware SDK.
-        return data.Select(e => new EmissionsData() 
+        return gridEmissionDataPoints.Select(e => new EmissionsData() 
                     { 
                         Location = e.BalancingAuthorityAbbreviation, 
                         Rating = ConvertMoerToGramsPerKilowattHour(e.Value), 
                         Time = e.PointTime,
-                        Duration = FrequencyToTimeSpan(e.Frequency)
+                        Duration = FrequencyToTimeSpanOrDefault(e.Frequency, defaultDuration)
                     });
     }
 
-    private TimeSpan GetDurationFromGridEmissionDataPoints(GridEmissionDataPoint? firstPoint, GridEmissionDataPoint? secondPoint)
+    private TimeSpan GetDurationFromGridEmissionDataPoints(IEnumerable<GridEmissionDataPoint> gridEmissionDataPoints)
     {
+        var firstPoint = gridEmissionDataPoints.FirstOrDefault(); 
+        var secondPoint = gridEmissionDataPoints.Skip(1)?.FirstOrDefault();
```

Comment:
Just for my own knowledge, if the array is empty, the `.Skip(1)` call won't crash, just goes to null as if there was only 1 element as well?

Reply:
Correct, but I will add a test case to ensure.

Reply:
Might it be easier to just check in this call in the array is at least 2 elements long?

Reply:
> Just for my own knowledge, if the array is empty, the `.Skip(1)` call won't crash, just goes to null as if there was only 1 element as well?

Same concern here. How about the 1st record?

Reply:
The reason for this approach is that it precisely accesses only the datapoints required for calculation, whereas `Count` is going to iterate through the entire enumerable.

Given that this approach is a direct port of the current implementation, we will likely leave it as is unless someone feels strongly about the readability.

Reply:
@juzuluag @JenMadiedo Here are the test cases exercising 0, 1, and 2 element responses through this code path.  https://github.com/microsoft/carbon-aware-sdk/pull/117/files#diff-248efed1bc704787aa3472bf81b17048d1a99d755c9b73e107da03163071b99aR258-R262